qemu/tests/win_video_play.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import time
+
+from autotest.client import utils
+from avocado.core import exceptions
+from virttest import data_dir
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+
+    def check_wmplayer_installed(session):
+        """
+        Check if wmplayer is installed
+        """
+        error_context.context("Check if wmplayer is installed", logging.info)
+        install_path = params.get("wmplayer_path")
+        check_cmd = 'dir "%s"|findstr /I wmplayer'
+        check_cmd = params.get("wmplayer_check_cmd", check_cmd) % install_path
+        if session.cmd_status(check_cmd) == 0:
+            return install_path
+
+    def check_kmplayer_installed(session, vm, tmp_dir):
+        """
+        Check if kmplayer is installed
+        """
+        error_context.context("Check if kmplayer is installed", logging.info)
+        install_path = params.get("kmplayer_path")
+        check_cmd = 'dir "%s"|findstr /I kmplayer'
+        check_cmd = params.get("kmplayer_check_cmd", check_cmd) % install_path
+        if session.cmd_status(check_cmd) != 0:
+            kmplayer_install(session, vm, tmp_dir)
+        return install_path
+
+    def kmplayer_install(session, vm, tmp_dir):
+        """
+        Install kmplayer
+        """
+        error_context.context("Install kmplayer ...", logging.info)
+        guest_name = params["guest_name"]
+        alias_map = params.get("guest_alias")
+        if alias_map:
+            guest_list = dict([x.split(":") for x in alias_map.split(";")])
+            print guest_list
+            if not guest_list[guest_name]:
+                raise exceptions.TestError("guest is not in guest_list,"
+                                           "pls check cfg file")
+            else:
+                folder_name = guest_list[guest_name].split(",")[0]
+                pkg_md5sum = guest_list[guest_name].split(",")[1]
+        src_url = params["kmplayer_source"] % folder_name
+        error_context.context("Download kmplayer source files", logging.info)
+        pkg_tmp = utils.unmap_url_cache(tmp_dir, src_url, pkg_md5sum)
+        error_context.context("Extract the pkg...", logging.info)
+        tar_cmd = params.get("tar_cmd", "tar xvf")
+        tar_cmd = "%s %s -C %s" % (tar_cmd, pkg_tmp, tmp_dir)
```

Comment:
import tarfile
t = tarfile.open('/path/pkg.tar', 'r:')
if it is gz file, could t = tarfile.open('/path/pkg.tar', 'r:gz')
t.extractall(path='extract_dir')
t.close
does this module meet your demand